kybra/compiler/kybra_generate/src/py_ast/node/canister_method/mod.rs

```diff
@@ -0,0 +1,102 @@
+pub mod errors;
+pub mod heartbeat_method;
+pub mod init_method;
+pub mod inspect_message_method;
+pub mod post_upgrade_method;
+pub mod pre_upgrade_method;
+mod query_or_update;
+
+use std::ops::Deref;
+
+use cdk_framework::act::node::canister_method::CanisterMethodType;
+use cdk_framework::act::node::param::Param;
+use proc_macro2::TokenStream;
+use rustpython_parser::ast::ExprKind;
+use rustpython_parser::ast::Located;
+use rustpython_parser::ast::Mod;
+use rustpython_parser::ast::StmtKind;
+
+use crate::generators::canister_methods;
+use crate::py_ast::PyAst;
+use crate::source_map::SourceMapped;
+
+pub use query_or_update::query_method;
+pub use query_or_update::update_method;
+
+impl PyAst {
+    fn get_canister_stmt_of_type(
+        &self,
+        method_type: CanisterMethodType,
+    ) -> Vec<SourceMapped<&Located<StmtKind>>> {
+        self.source_mapped_mods
+            .iter()
+            .fold(vec![], |mut acc, program| {
+                acc.extend(match &program.deref() {
+                    Mod::Module { body, .. } => body
+                        .iter()
+                        .filter(|stmt_kind| {
+                            SourceMapped::new(*stmt_kind, program.source_map.clone())
+                                .is_canister_method_type(method_type.clone())
+                        })
+                        .map(|stmt_kind| SourceMapped::new(stmt_kind, program.source_map.clone()))
+                        .collect(),
+                    _ => vec![],
+                });
+                acc
+            })
+    }
+}
+
+impl SourceMapped<&Located<StmtKind>> {
+    pub fn is_canister_method_type(&self, canister_method_type: CanisterMethodType) -> bool {
+        self.is_decorator_same_as(match canister_method_type {
+            CanisterMethodType::Heartbeat => "heartbeat",
+            CanisterMethodType::Init => "init",
+            CanisterMethodType::InspectMessage => "inspect_message",
+            CanisterMethodType::PostUpgrade => "post_upgrade",
+            CanisterMethodType::PreUpgrade => "pre_upgrade",
+            CanisterMethodType::Query => "query",
+            CanisterMethodType::Update => "update",
+        })
+    }
+
+    fn is_decorator_same_as(&self, decorator_name: &str) -> bool {
+        match &self.node {
+            StmtKind::FunctionDef { decorator_list, .. } => {
+                decorator_list
+                    .iter()
+                    .any(|expr_kind| match &expr_kind.node {
+                        ExprKind::Name { id, .. } => id == decorator_name,
+                        ExprKind::Call { func, .. } => match &func.node {
+                            ExprKind::Name { id, .. } => id == decorator_name,
+                            _ => false,
+                        },
+                        _ => false,
+                    })
+            }
+            _ => false,
+        }
+    }
+
+    pub fn build_params(&self) -> Vec<Param> {
```

Comment:
change build to to (ie to_params) Make sure "to" is for creation and get is for just simply getting, and as is for simple converting.

Reply:
All of the build stuff is looking good. Right now build is called when something doesn't exist, and needs to be built. If something can be converted to something then we are using to. But for example if a canister has some function decls that need to be made into canister methods, it wouldn't make sense to call to_canister_methods on the canister. And get I don't think make sense either because it doesn't have them yet they need to be built. I'm going to go through and make sure that to and get are being used consistently but build looks good.

Reply:
to and get look fine